TheWarWithin/HunterBeastMastery.lua

```diff
@@ -1573,20 +1589,30 @@ spec:RegisterAbilities( {
         gcd = "spell",
         school = "physical",
 
-        spend = 35,
+        spend = function () return talent.cobra_senses.enabled and 30 or 35 end,
         spendType = "focus",
 
         talent = "cobra_shot",
         startsCombat = true,
 
         handler = function ()
-            if debuff.concussive_shot.up then debuff.concussive_shot.expires = debuff.concussive_shot.expires + 3 end
 
-            if talent.killer_cobra.enabled and buff.bestial_wrath.up then setCooldown( "kill_command", 0 )
-            else
-                gainChargeTime( "kill_command", talent.cobra_senses.enabled and 2 or 1 )
+            if talent.serpentine_rhythm.enabled then
+                if buff.serpentine_rhythm.stacks == 3 then
```

Comment:
This means Serpentine Blessing procs when you'd reach 4 stacks, since there's no application of Serpenting Rhythm ahead of this point in the handler.  (That may be right, IDK.)

Reply:
Correct. You build up to 3, then the 4th cast removes it and you start over again at 0, no rhythm on your character anymore, but with blessing. Then casting again you'd still have blessing and get yourself 1 new stack of rhythm.